youboraplugin/src/main/java/com/kaltura/playkit/plugins/youbora/PKYouboraPlayerAdapter.java

```diff
@@ -133,15 +134,15 @@ private void sendErrorHandler(PKEvent event) {
         LinkedHashSet<String> causeMessages = getExceptionMessageChain(playerErrorException);
         StringBuilder exceptionCauseBuilder = new StringBuilder();
         if (playerErrorException != null && causeMessages.isEmpty()) {
-            exceptionCauseBuilder.append(playerErrorException.toString());
+            exceptionCauseBuilder.append(playerErrorException.toString()).append("\n");
         } else {
             for (String cause : causeMessages) {
                 exceptionCauseBuilder.append(cause).append("\n");
             }
         }
 
-        String errorCode = (errorEvent.error.errorType != null) ? errorEvent.error.errorType + " - " : "";
-        fireFatalError(exceptionCauseBuilder.toString(), errorCode + exceptionClass, errorMetadata);
+        String errorCode = (errorEvent.error.errorType != null) ? "" + ((PKPlayerErrorType) errorEvent.error.errorType).errorCode : "";
```

Comment:
@GouravSna  please check first instance of before casting  && that errorCode not null

Reply:
done